backend/server/tests/courses/test_get_course.py

```diff
@@ -29,11 +29,23 @@ def test_get_archived_course():
 
 
 def test_get_course_all_courses():
-    failed_courses = [
+    failed_courses = {
         course for course in CONDITIONS.keys()
         if (
-            random.random() < 0.5 and # Comment out this line to test all courses
+            # random.random() < 0.5 and # Comment out this line to test all courses
             requests.get(f'http://127.0.0.1:8000/courses/getCourse/{course}').status_code != 200
         )
-    ]
-    assert failed_courses == [], f"Total of {len(failed_courses)} courses failed.\n{failed_courses}"
+    }
+
+    # Offered alternate years => offered alternate years (not this year) but entry exists
+    # Not Offered => it is not offered, no further information, entry exists
+    known_failed_courses: set[str] = {
+        'PSCY9914', # Offered alternate years
+        'BEES3223', # Offered alternate years
+        'CEIC6714', # Not offered this year
+        'PSCY9901', # Offered alternate years
+        'PSCY9912', # Offered alternate years
+        'PHCM9612', # Not offered this year
+    }
+    failed_courses = failed_courses.difference(known_failed_courses)
```

Comment:
why do courses offered in alternate years fail?

Reply:
I'm actually not 100% sure. I think it might be a database thing? Because if you take a course like `PSYC9914`, it definitely exists inside `coursesProcessed.json` at line `50716`.

However, it seems that in the `getCourse` route, the following line:

```py
    result = coursesCOL.find_one({"code": courseCode})
```

returns `result` as `None`

Reply:
My bet is that you need to rebuild your `init-mongo` image

Reply:
Hmm possibly. Though then I'm unsure as to why it would have been failing on the pipeline though.